build_runner/lib/src/daemon/daemon_builder.dart

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:build/build.dart';
+import 'package:build_daemon/daemon_builder.dart';
+import 'package:build_daemon/data/build_status.dart' as daemon;
+import 'package:build_daemon/data/server_log.dart';
+import 'package:build_runner/src/entrypoint/options.dart';
+import 'package:build_runner/src/package_graph/build_config_overrides.dart';
+import 'package:build_runner/src/watcher/asset_change.dart';
+import 'package:build_runner/src/watcher/change_filter.dart';
+import 'package:build_runner/src/watcher/collect_changes.dart';
+import 'package:build_runner/src/watcher/delete_writer.dart';
+import 'package:build_runner/src/watcher/graph_watcher.dart';
+import 'package:build_runner/src/watcher/node_watcher.dart';
+import 'package:build_runner_core/build_runner_core.dart';
+import 'package:build_runner_core/src/generate/build_impl.dart';
+import 'package:logging/logging.dart';
+import 'package:watcher/watcher.dart';
+
+/// A Daemon Builder that uses build_runner_core for building.
+class BuildRunnerDaemonBuilder implements DaemonBuilder {
+  final _buildResults = StreamController<daemon.BuildResults>();
+  final _changesFromLastBuild = <AssetChange>[];
+
+  final BuildImpl _builder;
+  final BuildOptions _buildOptions;
+  final StreamController<ServerLog> _outputStreamController;
+  final Stream<AssetChange> _graphEvents;
+
+  BuildRunnerDaemonBuilder._(
+    this._builder,
+    this._buildOptions,
+    this._outputStreamController,
+    this._graphEvents,
+  );
+
+  @override
+  Stream<daemon.BuildResults> get builds => _buildResults.stream;
+
+  Stream<WatchEvent> get changes => _graphEvents.map((data) {
+        _changesFromLastBuild.add(data);
+        return WatchEvent(data.type, data.id.path);
+      });
+
+  @override
+  Stream<ServerLog> get logs => _outputStreamController.stream;
+
+  @override
+  Future<void> build(Set<String> targets, Set<String> logToPaths) async {
+    _logMessage(Level.INFO, 'About to build $targets...');
+    try {
+      var mergedChanges = collectChanges([_changesFromLastBuild]);
+      var result = await _builder.run(mergedChanges, targets.toList());
+      var results = <daemon.BuildResult>[];
+      for (var target in targets) {
+        if (result.status == BuildStatus.success) {
+          // TODO(grouma) - Can we notify if a target was cached?
```

Comment:
you could check if `result.outputs.isEmpty`?

Reply:
Discussed offline. We need to know if a given target is cached. This isn't a huge concern until Bolt onboards. I think there are alternatives.